addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverter.java

```diff
@@ -381,4 +388,137 @@ private void addOperations(OpenAPI openApi, String url, List<OperationModel> ope
+     * @return Populated either with a valid OpenAPI or a list of errors
+     */
+    public static SwaggerParseResult parse(File file) {
+        ParseOptions parseOptions = new ParseOptions();
+        parseOptions.setResolve(true);
+        parseOptions.setResolveFully(true);
+        SwaggerParseResult swaggerParseResult = new SwaggerParseResult();
+        List<String> errors = new ArrayList<>();
+        for (SwaggerParserExtension ex : OpenAPIV3Parser.getExtensions()) {
+            swaggerParseResult = ex.readLocation(file.getAbsolutePath(), null, parseOptions);
+            OpenAPI openAPI = swaggerParseResult.getOpenAPI();
+            if (openAPI != null && swaggerParseResult.getMessages().isEmpty()) {
+                return swaggerParseResult;
+            } else {
+                errors.addAll(swaggerParseResult.getMessages());
+            }
+        }
+        swaggerParseResult.setMessages(errors);
+        return swaggerParseResult;
+    }
+
+    public List<StructuralNodeModifier> getStructuralNodeModifiers() {
+        List<StructuralNodeModifier> structuralNodeModifiers = new ArrayList<>();
+        try {
+            OpenAPI openAPI = getOpenAPI();
+            String targetUrl;
+            if (targetUriBuilder.isEmpty()) {
+                targetUrl = null;
+            } else {
+                targetUrl = targetUriBuilder.build();
+            }
+
+            Set<String> ddnRegexStrings = new HashSet<>();
+            // check for DDN on each key (path) in the spec
+            for (Map.Entry<String, PathItem> entry : openAPI.getPaths().entrySet()) {
+                String key = entry.getKey();
+                try {
+                    List<String> params = getPathParameters(key);
+                    String ddnPattern = createStructuralNodeRegexString(params, targetUrl, key);
+                    if (ddnPattern != null) {
+                        ddnRegexStrings.add(ddnPattern);
+                    }
+                } catch (Exception e) {
+                    LOG.error("error evaluating DDN for key: {}", key, e);
+                }
+            }
+
+            int ddnCnt = 0;
+            for (String regexString : ddnRegexStrings) {
+                try {
+                    Pattern pattern = Pattern.compile(regexString);
+                    StructuralNodeModifier structuralNodeModifier =
+                            createStructuralNode(pattern, "DDN" + ddnCnt++);
+                    structuralNodeModifiers.add(structuralNodeModifier);
+                } catch (Exception e) {
+                    LOG.error("error adding structural node", e);
+                }
+            }
+
+        } catch (IllegalStateException | SwaggerException e) {
+            LOG.error("could not create structural nodes: {}", e.getMessage());
+        }
+        return structuralNodeModifiers;
+    }
+
+    public List<String> setStructuralNodeModifiers(int contextId) {
```

Comment:
Does not need to return the `errors` as it's already adding to the list of `errors`.